src/main/java/com/microsoft/aad/msal4j/TokenRequest.java

```diff
@@ -51,12 +52,15 @@
 class TokenRequest {
     Logger log = LoggerFactory.getLogger(TokenRequest.class);
 
-    private final URL url;
+    final Authority requestAuthority;
+    private final URL tokenEndpointUrl;
     private final MsalRequest msalRequest;
     private final ServiceBundle serviceBundle;
 
-    TokenRequest(final URL url, MsalRequest msalRequest, final ServiceBundle serviceBundle) {
-        this.url = url;
+    TokenRequest(Authority requestAuthority, MsalRequest msalRequest, ServiceBundle serviceBundle)
+            throws MalformedURLException {
+        this.requestAuthority = requestAuthority;
+        this.tokenEndpointUrl = new URL(requestAuthority.tokenEndpoint());
```

Comment:
Do you need both here, Perhaps authority.tokenEndpointUrl could do this instead?

Reply:
good point, changed